unstructured/ingest/connector/google_drive.py

```diff
@@ -0,0 +1,213 @@
+
+                self.file_meta.get("download_dir").mkdir(parents=True, exist_ok=True)
+
+            with open(self.filename, "wb") as handler:
+                handler.write(file.getbuffer())
+                if self.config.verbose:
+                    print(f"File downloaded: {self.filename}.")
+
+    def write_result(self):
+        """Write the structured json result for this doc. result must be json serializable."""
+        output_filename = self._output_filename()
+        output_filename.parent.mkdir(parents=True, exist_ok=True)
+        with open(output_filename, "w") as output_f:
+            output_f.write(json.dumps(self.isd_elems_no_filename, ensure_ascii=False, indent=2))
+        print(f"Wrote {output_filename}")
+
+
+class GoogleDriveConnector(BaseConnector):
+    """Objects of this class support fetching documents from Google Drive"""
+    def __init__(self, config):
+        self.config = config
+        self.cleanup_files = not self.config.preserve_downloads
+
+    def _list_objects(self, drive_id, recursive=False):
+        files = []
+
+        def traverse(drive_id, download_dir, output_dir, recursive=False):
+            page_token = None
+            while True:
+                response = self.config.service.files().list(
+                    spaces='drive', fields='nextPageToken, files(id, name, mimeType)',
+                    pageToken=page_token, corpora="user", q=f"'{drive_id}' in parents").execute()
+
+                for meta in response.get("files", []):
+                    if meta.get("mimeType") == "application/vnd.google-apps.folder":
+                        dir_ = DIRECTORY_FORMAT.format(name=meta.get("name"), id=meta.get("id"))
+                        if recursive:
+                            download_sub_dir = (download_dir / dir_).resolve()
+                            output_sub_dir = (output_dir / dir_).resolve()
+                            traverse(meta.get("id"), download_sub_dir, output_sub_dir, True)
+                    else:
+                        ext = ""
+                        if not Path(meta.get("name")).suffixes:
+                            guess = guess_extension(meta.get("mimeType"))
+                            ext = guess if guess else ext
+
+                        if meta.get("mimeType", "").startswith("application/vnd.google-apps"):
+                            export_mime = GOOGLE_DRIVE_EXPORT_TYPES.get(meta.get("mimeType"))
+                            if not export_mime:
+                                continue
+
+                            if not ext:
+                                guess = guess_extension(export_mime)
+                                ext = guess if guess else ext
+
+                        # TODO (Habeeb): Consider filtering at the query level.
+                        if self.config.extension and self.config.extension != ext:
+                            continue
```

Comment:
it be nice to print what we are skipping here (again)

Reply:
E.g., https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/connector/github.py does this nicely: both prints what is skipped and total number of files skipped.